lib/Image/Info/JPEG2000.pm

```diff
@@ -41,6 +49,21 @@ sub image_info {
         $$result{XResolution} = $$data{'jp2HeaderBox/resolutionBox/captureResolutionBox/hRescInPixelsPerInch'};
         $$result{YResolution} = $$data{'jp2HeaderBox/resolutionBox/captureResolutionBox/vRescInPixelsPerInch'};
         $$result{ResolutionUnit} = 'inches';
+    } elsif ( $$data{'uuidBox/xmpdata'} ) {
+        # there was no resolution data so pull it out of the XMP metadata
+        # print STDERR $$data{'uuidBox/xmpdata'}, "\n";
+        my $xmpdata = $$data{'uuidBox/xmpdata'};
+        require Image::TIFF;
+        foreach my $tag ( qw/XResolution YResolution ResolutionUnit/ ) {
+            my ( $tmp ) = ( $xmpdata =~ m,<tiff:$tag>(.*)</tiff:$tag>,m );
```

Comment:
I'm not wild about parsing the XMP with regular expressions. Will the namespace prefix actually always be `tiff`? Should we consider using `Image::Exiftool` instead of rolling our own thing here? 

For images in the repository, we should be able to always count on the JPEG2000 images having resolution info in the XMP rather than having to look at the JPEG2000 header.

Reply:
I'm not wild about it either; Image::ExifTool is orders of magnitude slower unfortunately and _some_ images in the repository do not have the resolution information available elsewhere. That's why the XML is only used if it's not available elsewhere.

Reply:
:+1: